spec/es_spec_helper.rb

```diff
@@ -34,14 +34,26 @@ def get_client
 RSpec.configure do |config|
   config.include ESHelper
 
-  config.before(:suite, :integration => true) do
-    @docker = Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
-    # TODO(talevy): detect when ES is up and ready instead of sleeping
-    # an arbitrary amount
-    sleep 10
+  config.before(:all, :integration => true) do
+    # check if container exists already
+    begin
+      ls = Longshoreman::new
+      ls.container.get(CONTAINER_NAME)
+    rescue Docker::Error::NotFoundError
+      Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
```

Comment:
to reduce instance variables / preserving state between spec runs, all docker state is saved in the docker environment externally. So we create the container, give it a name, and always search for it  from the docker-api before doing something with it